custom_components/octopus_energy/cost_tracker/cost_tracker.py

```diff
@@ -132,22 +135,25 @@ async def async_added_to_hass(self):
   async def _async_calculate_cost(self, event: EventType[EventStateChangedData]):
     new_state = event.data["new_state"]
     old_state = event.data["old_state"]
+    accumulative = event.data["new_state"].attributes.get(CONF_STATE_CLASS, SensorStateClass.MEASUREMENT) != SensorStateClass.MEASUREMENT
```

Comment:
For backwards compatability, it might be best to favour this if the state class is present. If it's not, then it should use the configuration value.

We could potentially mark the config value as deprecated with a link to a github issue to gather feedback if users are using cost trackers with sensors that don't include state classes.

This issue has highlighted more validation that I want to do when setting up the cost tracker sensors that should have been there from the beginning, but this would be another breaking change :(